Cannot build within GOPATH: build flag -mod=vendor only valid when using modules

Runing the build from within a GOPATH fails with:

```
:stackdriver-prometheus-sidecar (master)$ make docker
>> fetching promu
curl -s -L https://github.com/prometheus/promu/releases/download/v0.5.0/promu-0.5.0.linux-amd64.tar.gz | tar -xvzf - -C /tmp/tmp.zU4GpffJ6u
promu-0.5.0.linux-amd64/
promu-0.5.0.linux-amd64/promu
promu-0.5.0.linux-amd64/NOTICE
promu-0.5.0.linux-amd64/LICENSE
mkdir -p /home/rye/go/bin
cp /tmp/tmp.zU4GpffJ6u/promu-0.5.0.linux-amd64/promu /home/rye/go/bin/promu
rm -r /tmp/tmp.zU4GpffJ6u
>> building linux amd64 binaries
 >   stackdriver-prometheus-sidecar
build flag -mod=vendor only valid when using modules
!! command failed: build -o /home/rye/go/src/github.com/Stackdriver/stackdriver-prometheus-sidecar/stackdriver-prometheus-sidecar -ldflags -X github.com/Stackdriver/stackdriver-prometheus-sidecar/vendor/github.com/prometheus/common/version.Version=HEAD -X github.com/Stackdriver/stackdriver-prometheus-sidecar/vendor/github.com/prometheus/common/version.Revision=872c1b90aa8b2172f7aee766eeba5a3e921849b3 -X github.com/Stackdriver/stackdriver-prometheus-sidecar/vendor/github.com/prometheus/common/version.Branch=master -X github.com/Stackdriver/stackdriver-prometheus-sidecar/vendor/github.com/prometheus/common/version.BuildUser=rye@localhost -X github.com/Stackdriver/stackdriver-prometheus-sidecar/vendor/github.com/prometheus/common/version.BuildDate=20190827-22:07:23  -extldflags '-static' -mod=vendor -a -tags netgo github.com/Stackdriver/stackdriver-prometheus-sidecar/cmd/stackdriver-prometheus-sidecar: exit status 1
make: *** [Makefile:109: build-linux-amd64] Error 1
```

This appears to happen because `GO111MODULE`, while set in Makefile and used for `deps` target, is not propagated for other build steps.

--- a/Makefile
+++ b/Makefile
@@ -67,23 +67,23 @@ endif
 
 test-short:
 	@echo ">> running short tests"
-	$(GO) test -short $(GOOPTS) $(pkgs)
+	GO111MODULE=$(GO111MODULE) $(GO) test -short $(GOOPTS) $(pkgs)
 
 test:
 	@echo ">> running all tests"
-	$(GO) test $(GOOPTS) $(pkgs)
+	GO111MODULE=$(GO111MODULE) $(GO) test $(GOOPTS) $(pkgs)
 
 cover:
 	@echo ">> running all tests with coverage"
-	$(GO) test -coverprofile=coverage.out $(GOOPTS) $(pkgs)
+	GO111MODULE=$(GO111MODULE) $(GO) test -coverprofile=coverage.out $(GOOPTS) $(pkgs)
 
 format:
 	@echo ">> formatting code"
-	$(GO) fmt $(pkgs)
+	GO111MODULE=$(GO111MODULE) $(GO) fmt $(pkgs)
 
 vet:
 	@echo ">> vetting code"
-	$(GO) vet $(GOOPTS) $(pkgs)
+	GO111MODULE=$(GO111MODULE) $(GO) vet $(GOOPTS) $(pkgs)
 
 staticcheck: $(STATICCHECK)
 	@echo ">> running staticcheck"
@@ -98,11 +98,11 @@ endif
 
 build: promu
 	@echo ">> building binaries"
-	$(PROMU) build --prefix $(PREFIX)
+	GO111MODULE=$(GO111MODULE) $(PROMU) build --prefix $(PREFIX)
 
 build-linux-amd64: promu
 	@echo ">> building linux amd64 binaries"
-	@GOOS=linux GOARCH=amd64 $(PROMU) build --prefix $(PREFIX)
+	GO111MODULE=$(GO111MODULE) @GOOS=linux GOARCH=amd64 $(PROMU) build --prefix $(PREFIX)
 
 tarball: promu
 	@echo ">> building release tarball"

